fv3core/stencils/remap_profile.py

```diff
@@ -469,158 +467,207 @@ def set_inner(
-            ext6,
-            qmin,
-            abs(kord),
-            iv,
-            origin=(i1, j1, 2),
-            domain=(i_extent, j_extent, km - 4),
+        self._apply_constraints_stencil = gtscript.stencil(
+            definition=apply_constraints,
+            externals={"iv": iv, "kord": abs(kord)},
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
+        )
+
+        self._set_top_stencil = gtscript.stencil(
+            definition=set_top,
+            externals={"iv": iv},
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
+        )
+
+        self._set_set_inner_stencil = gtscript.stencil(
+            definition=set_inner,
+            externals={"iv": iv, "kord": abs(kord)},
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
+        )
+
+        self._set_bottom_stencil = gtscript.stencil(
+            definition=set_bottom,
+            externals={"iv": iv},
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
         )
 
-        set_bottom(
+    def __call__(
+        self,
+        qs: FloatField,
+        a4_1: FloatField,
+        a4_2: FloatField,
+        a4_3: FloatField,
+        a4_4: FloatField,
+        delp: FloatField,
+        i1: int,
+        i2: int,
+        j1: int,
+        j2: int,
+        qmin: float = 0.0,
+    ):
+        """
+        Calculates the interpolation coefficients for a cubic-spline which models the
+        distribution of the remapped field within each deformed grid cell.
+        The constraints on the spline are set by kord and iv.
+        Arguments:
+            qs: The field to be remapped
+            a4_1: The first interpolation coefficient
+            a4_2: The second interpolation coefficient
+            a4_3: The third interpolation coefficient
+            a4_4: The fourth interpolation coefficient
+            delp: The pressure difference between grid levels
+            i1: The first i-element to compute on
+            i2: The last i-element to compute on
+            j1: The first j-element to compute on
+            j2: The last j-element to compute on
+            qmin: The minimum value the field can take in a cell
+        """
+        i_extent: int = i2 - i1 + 1
+        j_extent: int = j2 - j1 + 1
+        orig: Tuple[int] = (i1, j1, 0)
+        dom: Tuple[int] = (i_extent, j_extent, self._km)
+        dom_extend: Tuple[int] = (i_extent, j_extent, self._km + 1)
+
+        self._set_values_stencil(
+            self._gam,
+            self._q,
+            delp,
             a4_1,
             a4_2,
             a4_3,
             a4_4,
-            extm,
-            iv,
-            origin=(i1, j1, km - 2),
-            domain=(i_extent, j_extent, 2),
+            self._q_bot,
+            qs,
+            origin=orig,
+            domain=dom_extend,
```

Comment:
potentially in a future version we might be able to identify the variations if i1/i2/j1/j2 that exist and make a different object for each, then compile the stencils using the  FixedOriginStencil, giving the origin and domain in compile time, reducing the time spend in normalize_origin and normalize_domain. But the impact may be small, and can be a future exploration. Making this an object is a huge and great step, and that would make sense as a another PR.

Reply:
In principle that makes sense, but right now the only place where the i and j extents aren't the compute domain are in the test code, I believe? Regardless, I agree that that probably makes sense as a separate PR